### PR TITLE
Fix tf random token masking probability in data collator

### LINKED ISSUE #21803: Masking ratio incorrect for DataCollatorForLanguageModeling
### System Info

- `transformers` version: 4.26.0
- Platform: Linux-4.19.0-23-cloud-amd64-x86_64-with-debian-10.13
- Python version: 3.7.12
- Huggingface_hub version: 0.12.0
- PyTorch version (GPU?): 1.13.1+cu116 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: No 
- Using distributed or parallel set-up in script?: No

### Who can help?

@sgugger
@ArthurZucker 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

It seems that there is an error in the implementation of the [torch_mask_tokens()](https://github.com/huggingface/transformers/blob/v4.26.1/src/transformers/data/data_collator.py#L750) method in the [DataCollatorForLanguageModeling](https://github.com/huggingface/transformers/blob/v4.26.1/src/transformers/data/data_collator.py#L609) class. 

According to the documentation, 80% of tokens are masked, 10% being replaced with the original tokens and 10% with random tokens. <s>However, the implementation for replacing tokens with random ones sets the probability at 50% instead of the intended 10%.</s>

``` python
indices_random = torch.bernoulli(torch.full(labels.shape, 0.5)).bool() & masked_indices & ~indices_replaced
random_words = torch.randint(len(self.tokenizer), labels.shape, dtype=torch.long)
inputs[indices_random] = random_words[indices_random]
```

EDIT: After a second look, it seems the logic here is that 80% tokens are masked, out of the remaining 20% tokens, half of them need to be replaced by random tokens.  So the probability 0.5 is correct. However, for [tf_mask_tokens()](https://github.com/huggingface/transformers/blob/v4.26.1/src/transformers/data/data_collator.py#L661), the probability is 0.1, which seems incorrect. Let me know if I understand it correctly!

``` python
indices_random = self.tf_bernoulli(input_shape, 0.1) & masked_indices & ~indices_replaced
random_words = tf.random.uniform(input_shape, maxval=vocab_size, dtype=tf.int64)
inputs = tf.where(indices_random, random_words, inputs)
```   

See the [link](https://github.com/huggingface/transformers/blob/v4.26.1/src/transformers/data/data_collator.py#L776) here.

Furthermore, I believe that the ratios for masked/original/random tokens should be configurable parameters that are accessible to users. At present, I have to inherit the `DataCollatorForLanguageModeling` class and override the `torch_mask_tokens` function in order to modify the ratio.

I can submit a pull request to address the bug and update the ratio parameter. Thank you very much!

### PR DESCRIPTION
**Pull request by anruijian**
# What does this PR do?

Fixes #21803 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

 @sgugger

### CODE REVIEW

**Review by ArthurZucker — APPROVED** [review]
thanks for fixing

**Comment by anruijian** [comment]
@ArthurZucker Do I need to merge it now?

**Review by sgugger — APPROVED** [review]
Thanks for the fix! Failure in the CI is unrelated and fixed on main, so merging.
